post_office/utils.py

```diff
@@ -1,159 +1,167 @@
+        * Value - file-like object, or a filename to open OR a dict of {'file': file-like-object, 'mimetype': string}
+
+    Returns a list of Attachment objects
+    """
+    attachments = []
+    for filename, filedata in attachment_files.items():
+
+        if isinstance(filedata, dict):
+            content = filedata.get('file', None)
+            mimetype = filedata.get('mimetype', None)
+            headers = filedata.get('headers', None)
+        else:
+            content = filedata
+            mimetype = None
+            headers = None
+
+        opened_file = None
+
+        if isinstance(content, str):
+            # `content` is a filename - try to open the file
+            opened_file = open(content, 'rb')
+            content = File(opened_file)
+
+        attachment = Attachment()
+        if mimetype:
+            attachment.mimetype = mimetype
+        attachment.headers = headers
+        attachment.name = filename
+        attachment.file.save(filename, content=content, save=True)
+
+        attachments.append(attachment)
+
+        if opened_file is not None:
+            opened_file.close()
+
+    return attachments
+
+
+def parse_priority(priority):
+    if priority is None:
+        priority = get_default_priority()
+    # If priority is given as a string, returns the enum representation
+    if isinstance(priority, str):
+        priority = getattr(PRIORITY, priority, None)
+
+        if priority is None:
+            raise ValueError('Invalid priority, must be one of: %s' %
+                             ', '.join(PRIORITY._fields))
+    return priority
+
+
+def parse_emails(emails):
+    """
+    A function that returns a list of valid email addresses.
+    This function will also convert a single email address into
+    a list of email addresses.
+    None value is also converted into an empty list.
+    """
+
+    if isinstance(emails, str):
+        emails = [emails]
+    elif emails is None:
+        emails = []
+
+    for email in emails:
+        try:
+            validate_email_with_name(email)
+        except ValidationError:
+            raise ValidationError('%s is not a valid email address' % email)
+
+    return emails
+
+
+def cleanup_expired_mails(cutoff_date, delete_attachments=True, batch_size=1000):
+    """
+    Delete all emails before the given cutoff date.
+    Optionally also delete pending attachments.
+    Return the number of deleted emails and attachments.
+    """
+    expired_emails = Email.objects.only('id').filter(created__lt=cutoff_date).values_list('id', flat=True)
+    total_deleted_emails = 0
+    while expired_emails:
+        email_ids = expired_emails[:batch_size]
+        
+        # Delete email and incr total_deleted_emails counter
+        _, deleted_data = Email.objects.filter(id__in=email_ids).delete()
+        total_deleted_emails += deleted_data['post_office.Email']
+        expired_emails = Email.objects.filter(created__lt=cutoff_date)\

```

Comment:
Since we already pulled out all the expired IDs in line 147, we shouldn't need to do another query here.